### PR TITLE
Add delete as a multiple action

### LINKED ISSUE #101: Improve notification page
- [x] When deleting a notification modal dialog displays wrong title and body
- [x] What do we do with bulk actions? Do we add 'delete' to bulk actions?

### PR DESCRIPTION
**Pull request by SferaDev**
### :pushpin: References

* **Issue:** Closes #101 

### :memo: Implementation

- Update strings

- Allow multiple delete on Notifications and Instances

### :art: Screenshots

image

image

image

### :fire: Is there anything the reviewer should know to test it?

- The first proposal to fix this was executing concurrent updates to the dataStore but it had data races and did not delete all entries.

- We are now running the calls synchronous instead.

### CODE REVIEW

File: src/components/notifications-table/NotificationsTable.jsx [hunk 1]
```diff
@@ -41,35 +41,36 @@ class NotificationsTable extends React.Component {
...
-                );
-            }
-        });
+        loading.show(true, i18n.t("Deleting Notifications"));
+        const notifications = toDelete.map(data => new SyncReport(data));
+
+        const results = [];
+        for (const notification of notifications) {
+            results.push(await notification.remove(d2));
+        }
```
Comment (tokland) [inline src/components/notifications-table/NotificationsTable.jsx]: Whenever we have a pmap-like abstraction at hand: `const results = await promiseMap(notifications, notification => notification.remove(d2);)`, I think there is none now in the projcet.

File: src/components/instance-configurator/InstanceConfigurator.jsx [hunk 2]
```diff
@@ -46,35 +47,36 @@ class InstanceConfigurator extends React.Component {
...
+        loading.reset();
+        this.setState({ tableKey: Math.random(), toDelete: null });
+
+        if (_.some(results, ["status", false])) {
+            this.props.snackbar.error(i18n.t("Failed to delete some instances"));
+        } else {
+            this.props.snackbar.success(
+                i18n.t("Successfully deleted {{count}} instances", { count: toDelete.length })
+            );
+        }
```
[the remark this answers is not in the document]
Comment (SferaDev) [reply src/components/instance-configurator/InstanceConfigurator.jsx]: Thought of that too when redoing both, sounds good to me.

File: src/components/notifications-table/NotificationsTable.jsx (hunk 1 above)
Comment (SferaDev) [reply src/components/notifications-table/NotificationsTable.jsx]: Yeah looks good when we implement race protection on data store level.

**Comment by adrianq** [comment]
@SferaDev All good, I just committed this minor change in the yarn.lock https://github.com/EyeSeeTea/metadata-synchronization/pull/102/commits/daa63e59abf182dcb4ac8e3081b55237bbfcc689 It looks like that dep is not needed anymore.

**Pull request merged**